### ISSUE TITLE
[Bug] Visual errors rendering when typing in text boxes on Firefox 84+ (Normal/Beta/Nightly)

### ISSUE THREAD

**Issue by Tanebi** [mozilla-mobile/fenix#16969]
[comment]: # (Please do your best to search for duplicate issues before filing a new issue so we can keep our issue board clean)
[comment]: # (Every issue should have the exact bug and steps to reproduce described in it. Please do not file feedback list tickets as it is difficult to parse them and address their individual points)
[comment]: # (Read https://github.com/mozilla-mobile/fenix#i-want-to-file-an-issue for more information)

## Steps to reproduce
On Samsung Note20 Ultra update Firefox to v84, or install Nightly/Beta versions 84+.  
Go to Google.com
Type text in search area on page.

### Expected behavior
Text appears and is correct, page continues to function as normal. Clicking enter goes to search with expected value.

### Actual behavior
Text is corrupted/garbled and random text is seen. Clicking enter goes to search with expected search value  
Image of results of typing "hello" https://i.imgur.com/YPBdcDN.jpg

### Device information

* Android device: Samsung Note20 Ultra
* Fenix version: 84
 
On Firefox Beta navigating to `about:config` and setting `gfx.webrender.force-disabled` to `true` causes the browser to fall back to OpenGL compositing and correct rendering of the page. 

about:config is not available on "Standard" Firefox Android.

Output of about:support follows.

> Application Basics
> ------------------
> 
> Name: Firefox Beta
> Version: 84.0b9
> Build ID: 20201206192040
> Distribution ID:
> User Agent: Mozilla/5.0 (Android 10; Mobile; rv:84.0) Gecko/84.0 Firefox/84.0
> OS: Linux 29 #1 SMP PREEMPT Mon Nov 23 21:34:27 KST 2020
> Multiprocess Windows: 0/0
> Fission Windows: 0/0 Disabled by default
> Remote Processes: 1
> Google Location Service Key: Found
> Google Safebrowsing Key: Found
> Mozilla Location Service Key: Found
> Safe Mode: false
> 
> Crash Reports for the Last 3 Days
> ---------------------------------
> 
> Nightly Features
> ----------------
> 
> Remote Processes
> ----------------
> 
> Type: Web Content
> Count: 1 / 1
> 
> Add-ons
> -------
> 
> Name: Mozilla Android Components - Ads
> Type: extension
> Version: 67.0.9.1207225947
> Enabled: true
> ID: ads@mozac.org
> 
> Name: Mozilla Android Components - Browser Icons
> Type: extension
> Version: 67.0.9.1207141154
> Enabled: true
> ID: icons@mozac.org
> 
> Name: Mozilla Android Components - Cookies
> Type: extension
> Version: 67.0.9.1207225947
> Enabled: true
> ID: cookies@mozac.org
> 
> Name: Mozilla Android Components - Firefox Accounts WebChannel
> Type: extension
> Version: 67.0.9.1207140911
> Enabled: true
> ID: fxa@mozac.org
> 
> Name: Mozilla Android Components - ReaderView
> Type: extension
> Version: 67.0.9.1207141350
> Enabled: true
> ID: readerview@mozac.org
> 
> Name: Mozilla Android Components - Web Compatibility Interventions
> Type: extension
> Version: 17.0.0
> Enabled: true
> ID: webcompat@mozilla.org
> 
> Name: Mozilla Android Components - WebCompat Reporter
> Type: extension
> Version: 2.0.0
> Enabled: true
> ID: webcompat-reporter@mozilla.org
> 
> Graphics
> --------
> 
> Features
> Compositing: WebRender
> Asynchronous Pan/Zoom: touch input enabled; autoscroll enabled; smooth pinch-zoom enabled
> WebGL 1 Driver WSI Info: EGL_VENDOR: Android EGL_VERSION: 1.4 Android META-EGL EGL_EXTENSIONS: EGL_KHR_get_all_proc_addresses EGL_ANDROID_presentation_time EGL_KHR_swap_buffers_with_damage EGL_ANDROID_get_native_client_buffer EGL_ANDROID_front_buffer_auto_refresh EGL_ANDROID_get_frame_timestamps EGL_EXT_surface_SMPTE2086_metadata EGL_EXT_surface_CTA861_3_metadata EGL_EXT_gl_colorspace_scrgb EGL_EXT_gl_colorspace_scrgb_linear EGL_EXT_gl_colorspace_display_p3_linear EGL_EXT_gl_colorspace_display_p3 EGL_EXT_gl_colorspace_display_p3_passthrough EGL_EXT_gl_colorspace_bt2020_linear EGL_EXT_gl_colorspace_bt2020_pq EGL_KHR_image EGL_KHR_image_base EGL_EXT_image_gl_colorspace EGL_KHR_gl_colorspace EGL_KHR_gl_texture_2D_image EGL_KHR_gl_texture_cubemap_image EGL_KHR_gl_renderbuffer_image EGL_KHR_fence_sync EGL_KHR_create_context EGL_KHR_config_attribs EGL_KHR_surfaceless_context EGL_EXT_create_context_robustness EGL_ANDROID_image_native_buffer EGL_KHR_wait_sync EGL_ANDROID_recordable EGL_KHR_partial_update EGL_EXT_pixel_format_float EGL_KHR_mutable_render_buffer EGL_EXT_protected_content EGL_IMG_context_priority EGL_KHR_no_config_context IsWebglOutOfProcessEnabled: 0
> WebGL 1 Driver Renderer: ARM -- Mali-G77
> WebGL 1 Driver Version: OpenGL ES 3.2 v1.r20p0-01rel0.###other-sha0123456789ABCDEF0###
> WebGL 1 Driver Extensions: GL_EXT_debug_marker GL_ARM_rgba8 GL_ARM_mali_shader_binary GL_OES_depth24 GL_OES_depth_texture GL_OES_depth_texture_cube_map GL_OES_packed_depth_stencil GL_OES_rgb8_rgba8 GL_EXT_read_format_bgra GL_OES_compressed_paletted_texture GL_OES_compressed_ETC1_RGB8_texture GL_OES_standard_derivatives GL_OES_EGL_image GL_OES_EGL_image_external GL_OES_EGL_image_external_essl3 GL_OES_EGL_sync GL_OES_texture_npot GL_OES_vertex_half_float GL_OES_required_internalformat GL_OES_vertex_array_object GL_OES_mapbuffer GL_EXT_texture_format_BGRA8888 GL_EXT_texture_rg GL_EXT_texture_type_2_10_10_10_REV GL_OES_fbo_render_mipmap GL_OES_element_index_uint GL_EXT_shadow_samplers GL_OES_texture_compression_astc GL_KHR_texture_compression_astc_ldr GL_KHR_texture_compression_astc_hdr GL_KHR_texture_compression_astc_sliced_3d GL_EXT_texture_compression_astc_decode_mode GL_EXT_texture_compression_astc_decode_mode_rgb9e5 GL_KHR_debug GL_EXT_occlusion_query_boolean GL_EXT_disjoint_timer_query GL_EXT_blend_minmax GL_EXT_discard_framebuffer GL_OES_get_program_binary GL_OES_texture_3D GL_EXT_texture_storage GL_EXT_multisampled_render_to_texture GL_EXT_multisampled_render_to_texture2 GL_OES_surfaceless_context GL_OES_texture_stencil8 GL_EXT_shader_pixel_local_storage GL_ARM_shader_framebuffer_fetch GL_ARM_shader_framebuffer_fetch_depth_stencil GL_ARM_mali_program_binary GL_EXT_sRGB GL_EXT_sRGB_write_control GL_EXT_texture_sRGB_decode GL_EXT_texture_sRGB_R8 GL_EXT_texture_sRGB_RG8 GL_KHR_blend_equation_advanced GL_KHR_blend_equation_advanced_coherent GL_OES_texture_storage_multisample_2d_array GL_OES_shader_image_atomic GL_EXT_robustness GL_EXT_draw_buffers_indexed GL_OES_draw_buffers_indexed GL_EXT_texture_border_clamp GL_OES_texture_border_clamp GL_EXT_texture_cube_map_array GL_OES_texture_cube_map_array GL_OES_sample_variables GL_OES_sample_shading GL_OES_shader_multisample_interpolation GL_EXT_shader_io_blocks GL_OES_shader_io_blocks GL_EXT_tessellation_shader GL_OES_tessellation_shader GL_EXT_primitive_bounding_box GL_OES_primitive_bounding_box GL_EXT_geometry_shader GL_OES_geometry_shader GL_ANDROID_extension_pack_es31a GL_EXT_gpu_shader5 GL_OES_gpu_shader5 GL_EXT_texture_buffer GL_OES_texture_buffer GL_EXT_copy_image GL_OES_copy_image GL_EXT_shader_non_constant_global_initializers GL_EXT_color_buffer_half_float GL_EXT_color_buffer_float GL_EXT_YUV_target GL_OVR_multiview GL_OVR_multiview2 GL_OVR_multiview_multisampled_render_to_texture GL_KHR_robustness GL_KHR_robust_buffer_access_behavior GL_EXT_draw_elements_base_vertex GL_OES_draw_elements_base_vertex GL_EXT_protected_textures GL_EXT_buffer_storage GL_EXT_external_buffer GL_EXT_EGL_image_array GL_EXT_texture_filter_anisotropic GL_OES_texture_float_linear
> WebGL 1 Extensions: ANGLE_instanced_arrays EXT_blend_minmax EXT_color_buffer_half_float EXT_float_blend EXT_sRGB EXT_texture_filter_anisotropic MOZ_debug OES_element_index_uint OES_fbo_render_mipmap OES_standard_derivatives OES_texture_float OES_texture_float_linear OES_texture_half_float OES_texture_half_float_linear OES_vertex_array_object WEBGL_color_buffer_float WEBGL_compressed_texture_astc WEBGL_compressed_texture_etc WEBGL_compressed_texture_etc1 WEBGL_debug_renderer_info WEBGL_debug_shaders WEBGL_depth_texture WEBGL_lose_context
> WebGL 2 Driver WSI Info: EGL_VENDOR: Android EGL_VERSION: 1.4 Android META-EGL EGL_EXTENSIONS: EGL_KHR_get_all_proc_addresses EGL_ANDROID_presentation_time EGL_KHR_swap_buffers_with_damage EGL_ANDROID_get_native_client_buffer EGL_ANDROID_front_buffer_auto_refresh EGL_ANDROID_get_frame_timestamps EGL_EXT_surface_SMPTE2086_metadata EGL_EXT_surface_CTA861_3_metadata EGL_EXT_gl_colorspace_scrgb EGL_EXT_gl_colorspace_scrgb_linear EGL_EXT_gl_colorspace_display_p3_linear EGL_EXT_gl_colorspace_display_p3 EGL_EXT_gl_colorspace_display_p3_passthrough EGL_EXT_gl_colorspace_bt2020_linear EGL_EXT_gl_colorspace_bt2020_pq EGL_KHR_image EGL_KHR_image_base EGL_EXT_image_gl_colorspace EGL_KHR_gl_colorspace EGL_KHR_gl_texture_2D_image EGL_KHR_gl_texture_cubemap_image EGL_KHR_gl_renderbuffer_image EGL_KHR_fence_sync EGL_KHR_create_context EGL_KHR_config_attribs EGL_KHR_surfaceless_context EGL_EXT_create_context_robustness EGL_ANDROID_image_native_buffer EGL_KHR_wait_sync EGL_ANDROID_recordable EGL_KHR_partial_update EGL_EXT_pixel_format_float EGL_KHR_mutable_render_buffer EGL_EXT_protected_content EGL_IMG_context_priority EGL_KHR_no_config_context IsWebglOutOfProcessEnabled: 0
> WebGL 2 Driver Renderer: ARM -- Mali-G77
> WebGL 2 Driver Version: OpenGL ES 3.2 v1.r20p0-01rel0.###other-sha0123456789ABCDEF0###
> WebGL 2 Driver Extensions: GL_EXT_debug_marker GL_ARM_rgba8 GL_ARM_mali_shader_binary GL_OES_depth24 GL_OES_depth_texture GL_OES_depth_texture_cube_map GL_OES_packed_depth_stencil GL_OES_rgb8_rgba8 GL_EXT_read_format_bgra GL_OES_compressed_paletted_texture GL_OES_compressed_ETC1_RGB8_texture GL_OES_standard_derivatives GL_OES_EGL_image GL_OES_EGL_image_external GL_OES_EGL_image_external_essl3 GL_OES_EGL_sync GL_OES_texture_npot GL_OES_vertex_half_float GL_OES_required_internalformat GL_OES_vertex_array_object GL_OES_mapbuffer GL_EXT_texture_format_BGRA8888 GL_EXT_texture_rg GL_EXT_texture_type_2_10_10_10_REV GL_OES_fbo_render_mipmap GL_OES_element_index_uint GL_EXT_shadow_samplers GL_OES_texture_compression_astc GL_KHR_texture_compression_astc_ldr GL_KHR_texture_compression_astc_hdr GL_KHR_texture_compression_astc_sliced_3d GL_EXT_texture_compression_astc_decode_mode GL_EXT_texture_compression_astc_decode_mode_rgb9e5 GL_KHR_debug GL_EXT_occlusion_query_boolean GL_EXT_disjoint_timer_query GL_EXT_blend_minmax GL_EXT_discard_framebuffer GL_OES_get_program_binary GL_OES_texture_3D GL_EXT_texture_storage GL_EXT_multisampled_render_to_texture GL_EXT_multisampled_render_to_texture2 GL_OES_surfaceless_context GL_OES_texture_stencil8 GL_EXT_shader_pixel_local_storage GL_ARM_shader_framebuffer_fetch GL_ARM_shader_framebuffer_fetch_depth_stencil GL_ARM_mali_program_binary GL_EXT_sRGB GL_EXT_sRGB_write_control GL_EXT_texture_sRGB_decode GL_EXT_texture_sRGB_R8 GL_EXT_texture_sRGB_RG8 GL_KHR_blend_equation_advanced GL_KHR_blend_equation_advanced_coherent GL_OES_texture_storage_multisample_2d_array GL_OES_shader_image_atomic GL_EXT_robustness GL_EXT_draw_buffers_indexed GL_OES_draw_buffers_indexed GL_EXT_texture_border_clamp GL_OES_texture_border_clamp GL_EXT_texture_cube_map_array GL_OES_texture_cube_map_array GL_OES_sample_variables GL_OES_sample_shading GL_OES_shader_multisample_interpolation GL_EXT_shader_io_blocks GL_OES_shader_io_blocks GL_EXT_tessellation_shader GL_OES_tessellation_shader GL_EXT_primitive_bounding_box GL_OES_primitive_bounding_box GL_EXT_geometry_shader GL_OES_geometry_shader GL_ANDROID_extension_pack_es31a GL_EXT_gpu_shader5 GL_OES_gpu_shader5 GL_EXT_texture_buffer GL_OES_texture_buffer GL_EXT_copy_image GL_OES_copy_image GL_EXT_shader_non_constant_global_initializers GL_EXT_color_buffer_half_float GL_EXT_color_buffer_float GL_EXT_YUV_target GL_OVR_multiview GL_OVR_multiview2 GL_OVR_multiview_multisampled_render_to_texture GL_KHR_robustness GL_KHR_robust_buffer_access_behavior GL_EXT_draw_elements_base_vertex GL_OES_draw_elements_base_vertex GL_EXT_protected_textures GL_EXT_buffer_storage GL_EXT_external_buffer GL_EXT_EGL_image_array GL_EXT_texture_filter_anisotropic GL_OES_texture_float_linear
> WebGL 2 Extensions: EXT_color_buffer_float EXT_float_blend EXT_texture_filter_anisotropic MOZ_debug OES_texture_float_linear OVR_multiview2 WEBGL_compressed_texture_astc WEBGL_compressed_texture_etc WEBGL_compressed_texture_etc1 WEBGL_debug_renderer_info WEBGL_debug_shaders WEBGL_lose_context
> Uses Tiling: true
> Uses Tiling (Content): true
> Target Frame Rate: 120
> GPU #1
> Active: Yes
> Description: Model: SM-N986B, Product: c2sxeea, Manufacturer: samsung, Hardware: exynos990, OpenGL: ARM -- Mali-G77 -- OpenGL ES 3.2 v1.r20p0-01rel0.###other-sha0123456789ABCDEF0###
> Vendor ID: ARM
> Device ID: Mali-G77
> Driver Version: OpenGL ES 3.2 v1.r20p0-01rel0.###other-sha0123456789ABCDEF0###
> RAM: 0
> 
> Diagnostics
> AzureCanvasBackend: skia
> AzureContentBackend: skia
> AzureFallbackCanvasBackend: none
> CMSOutputProfile: Empty profile data
> Display0: 1080x2316
> DisplayCount: 1
> TileHeight: 512
> TileWidth: 512
> Decision Log
> HW_COMPOSITING:
> available by default
> force_enabled by user: Force-enabled by pref
> OPENGL_COMPOSITING:
> available by default
> force_enabled by user: Force-enabled by pref
> WEBRENDER:
> available by default
> force_enabled by user: Force enabled by pref
> WEBRENDER_QUALIFIED:
> available by default
> WEBRENDER_COMPOSITOR:
> disabled by default: Disabled by default
> WEBRENDER_PARTIAL:
> available by default
> WEBRENDER_ANGLE:
> available by default
> unavailable by env: OS not supported
> WEBRENDER_DCOMP_PRESENT:
> available by default
> disabled by user: User disabled via pref
> unavailable by env: Requires Windows 10 or later
> unavailable by runtime: Requires ANGLE
> WEBRENDER_SOFTWARE:
> available by default
> disabled by env: Overriden by qualified hardware
> OMTP:
> disabled by default: Disabled by default
> WEBGPU:
> disabled by default: Disabled by default
> blocked by runtime: WebGPU can only be enabled in nightly
> 
> 
> 
> 
> Media
> -----
> 
> Audio Backend: opensl
> Max Channels: 2
> Preferred Sample Rate: 48000
> Roundtrip latency (standard deviation): ...
> Output Devices
> Name: Group
> :
> Input Devices
> Name: Group
> :
> 
> Enumerate database
> 
> Environment Variables
> ---------------------
> 
> Important Modified Preferences
> ------------------------------
> 
> dom.presentation.device.name: SM-N986B
> dom.w3c_pointer_events.enabled: true
> extensions.lastAppVersion: 84.0
> idle.lastDailyNotification: 1607598316
> media.gmp.storage.version.observed: 1
> network.trr.blocklist_cleanup_done: true
> privacy.purge_trackers.date_in_cookie_database: 0
> security.remote_settings.crlite_filters.checked: 1607598383
> security.remote_settings.intermediates.checked: 1607598383
> storage.vacuum.last.index: 0
> 
> Important Locked Preferences
> ----------------------------
> 
> fission.autostart: false
> fission.autostart.session: false
> 
> Accessibility
> -------------
> 
> Activated: true
> Prevent Accessibility: 0
> 
> Library Versions
> ----------------
> 
> NSPR
> Expected minimum version: 4.29
> Version in use: 4.29
> 
> NSS
> Expected minimum version: 3.59
> Version in use: 3.59
> 
> NSSSMIME
> Expected minimum version: 3.59
> Version in use: 3.59
> 
> NSSSSL
> Expected minimum version: 3.59
> Version in use: 3.59
> 
> NSSUTIL
> Expected minimum version: 3.59
> Version in use: 3.59
> 
> Startup Cache
> -------------
> 
> Disk Cache Path: /data/user/0/org.mozilla.firefox_beta/files/mozilla/e3sbmcp7.default/startupCache/startupCache.8.little
> Ignore Disk Cache: false
> Found Disk Cache on Init: false
> Wrote to Disk Cache: true
> 
> Internationalization & Localization
> -----------------------------------
> 
> Application Settings
> Requested Locales: ["en-GB"]
> Available Locales: ["an","ar","ast","az","be","bg","bn","br","bs","ca","cak","cs","cy","da","de","dsb","el","en-CA","en-GB","en-US","eo","es-AR","es-CL","es-ES","es-MX","et","eu","fa","ff","fi","fr","fy-NL","ga-IE","gd","gl","gn","gu-IN","he","hi-IN","hr","hsb","hu","hy-AM","id","is","it","ja","ka","kab","kk","kn","ko","lij","lo","lt","lv","ml","mr","ms","my","nb-NO","ne-NP","nl","nn-NO","oc","pa-IN","pl","pt-BR","pt-PT","rm","ro","ru","sk","sl","son","sq","sr","sv-SE","ta","te","th","tr","trs","uk","ur","uz","vi","wo","xh","zam","zh-CN","zh-TW"]
> App Locales: ["en-GB","en-US","en-CA"]
> Regional Preferences: ["en-GB"]
> Default Locale: "en-US"
> Operating System
> System Locales: ["en-GB"]
> Regional Preferences: ["en-GB"]

**Comment by Tanebi** [comment]
Duplicates https://github.com/mozilla-mobile/fenix/issues/13153#issuecomment-742159588
